include/eve/detail/shuffle_v2/shuffle_v2_driver_fwd.hpp

```diff
@@ -74,12 +96,19 @@ template<typename Internal> struct shuffle_reverse_arguments
     return kumi::apply(internal, kumi::reverse(kumi::tuple {args...}));
   }
 
-  template<typename... Ts>
   EVE_FORCEINLINE auto operator()(auto... args) const noexcept
   requires(matched_shuffle<decltype(get<0>(impl(args...)))>)
   {
     return impl(args...);
   }
+
+  EVE_FORCEINLINE auto operator()(auto... args) const noexcept
+  requires(shuffle_user_error<decltype(impl(args...))>)
+  {
+    auto error = impl(args...);
+    static_assert(!shuffle_user_error<decltype(error)>);
```

Comment:
this is a top level function, so the error is nice.
It's also not sfinaeble: you can sfinae on "shuffle not found" but you cannot on "I made an input error"